draft-irtf-cfrg-vdaf.md

```diff
@@ -3074,8 +3080,47 @@ def decode(self, output, _num_measurements):
     return [bucket_count.as_unsigned() for bucket_count in output]
 ~~~
 
-The `Histogram` validity circuit uses `Range2` (see {{prio3sum}}) as its single
-gadget. It checks for one-hotness in two steps, as follows:
+This validity circuit uses a `ParallelSum` gadget to achieve a smaller proof
+size. This optimization for "parallel-sum circuits" is described in
+{{BBCGGI19}}, section 4.4. Briefly, for circuits that add up the output of
+multiple identical subcircuits, it is possible to achieve smaller proof sizes
+(on the order of O(sqrt(n)) instead of O(n)) by packaging more than one such
+subcircuit into a gadget.
+
+The `ParallelSum` gadget is parameterized with an arithmetic subcircuit, and a
+`count` of how many times it evaluates that subcircuit. It takes in a list of
+inputs and passes them through to instances of the subcircuit in the same order.
+It returns the sum of the subcircuit outputs. Note that only the `ParallelSum`
+gadget itself, and not its subcircuit, participates in `FlpGeneric`'s wire
+recording during evaluation, gadget consistency proofs, and proof validation,
+even though the subcircuit is provided to `ParallelSum` as an implementation
+of the `Gadget` interface.
```

Comment:
Thanks for moving this explanation here. However now that we have Prio3SumVec, I wonder if it should go right after Prio3Sum, which would mean moving this explanation back to Prio3SumVec 😮‍💨